3-dynamic-analysis/4-tracing/1-tracing.md

```diff
@@ -0,0 +1,166 @@
+# Tracing Go programs
+
+In Go 1.5, a new tool was added to Go: the execution tracer.
+In this chapter we will learn what the tracer does and how it can
+help us pin point performance issues in our programs.
+
+Unlike `pprof` which, as we saw before checks what a Go program is
+currently executing, the execution tracer makes the Go runtime
+take the initiative of reporting each time an event occurs.
+These events can be creation of goroutines, system calls, changes
+of heap size, and many more. Each time one of these events occurs
+it is reported with its timestamp and a stack trace for most events.
+
+The result of executing a program with the Go Execution Tracer enabled
+is a pretty big binary file that can be analyzed with  `go tool trace`.
+
+## A first example
+
+The awesome thing about the Go execution tracer is that it doesn't require
+to run for long, so we can start by understanding what a program does by
+simply adding calls to `trace.Start` and `trace.Stop`.
+
+[embedmd]:# (daisy.go /package main/ /trace.Stop/)
+```go
+package main
+
+import (
+	"fmt"
+	"os"
+	"runtime/trace"
+)
+
+func main() {
+	trace.Start(os.Stdout)
+	defer trace.Stop
+```
+
+So, without reading anything else in the code, let's simply run the code and
+store the trace output.
+
+```bash
+$ go run daisy.go > trace.out
+3
+$ go tool trace trace.out
+2017/07/10 17:47:47 Parsing trace...
+2017/07/10 17:47:47 Serializing trace...
+2017/07/10 17:47:47 Splitting trace...
+2017/07/10 17:47:47 Opening browser
+```
+
+This will open a browser with a serious of links, let's click on `Goroutine
```

Comment:
minor typo: it is _serious_, but I think you mean **series** of links. :joy_cat: 

By the way, love this whole series, but especially that you are adding this new section on `go tools trace`. Was trying to use it a year ago and other than digging in the source code, there is/was very little out there on this topic. So thank you!